test/aws_url.spec.js

```diff
@@ -0,0 +1,135 @@
+import chai, { expect } from 'chai'
+import chaiSinon from 'sinon-chai'
+import sinon from 'sinon'
+import test from 'ava'
+
+import Evaporate from '../evaporate'
+
+import initResponse from './fixtures/init-response'
+import completeResponse from './fixtures/complete-response'
+
+chai.use(chaiSinon)
+
+// consts
+
+const CONTENT_TYPE_XML = { 'Content-Type': 'text/xml' }
+const CONTENT_TYPE_TEXT = { 'Content-Type': 'text/plain' }
+
+const AWS_BUCKET = 'bucket'
+const AWS_UPLOAD_KEY = 'tests'
+
+const baseConfig = {
+  signerUrl: 'http://what.ever/sign',
+  aws_key: 'testkey',
+  bucket: AWS_BUCKET,
+  logging: false
+}
+
+const baseAddConfig = {
+  name: AWS_UPLOAD_KEY,
+  file: new File({
+    path: '/tmp/file',
+    size: 50
+  })
+}
+
+let server
+
+test.before(() => {
+  sinon.xhr.supportsCORS = true
+  server = sinon.fakeServer.create({
+    respondImmediately: true
+  })
+
+  server.respondWith('POST', /^.*\?uploads.*$/, (xhr) => {
+    xhr.respond(200, CONTENT_TYPE_XML, initResponse(AWS_BUCKET, AWS_UPLOAD_KEY))
+  })
+  server.respondWith('PUT', /^.*$/, (xhr) => {
+    xhr.respond(200)
+  })
+
+  server.respondWith('POST', /.*\?uploadId.*$/, (xhr) => {
+    xhr.respond(200, CONTENT_TYPE_XML, completeResponse(AWS_BUCKET, AWS_UPLOAD_KEY))
+  })
+
+  server.respondWith('GET', /\/sign.*$/, (xhr) => {
+    const payload = Array(29).join()
+    xhr.respond(200, CONTENT_TYPE_TEXT, payload)
+  })
+
+  server.respondWith('DELETE', /.*\?uploadId.*$/, (xhr) => {
+    xhr.respond(204)
+  })
+
+  global.XMLHttpRequest = sinon.fakeServer.xhr
+  global.setTimeout = (fc) => fc()
+})
+
+test.after(() => {
+  server.restore()
+})
+
+// test that AWS Url obeys the contract
+
+function testAwsUrl(obj, t, input, expected) {
+  const evapConfig = Object.assign({}, baseConfig, input)
+
+  const evaporate = new Evaporate(evapConfig)
+
+  const _handleUploadComplete = (xhr) => {
+    expect(xhr.url).to.include(expected)
+    t.end()
+  }
+
+  const config = Object.assign({}, baseAddConfig, {
+    complete: _handleUploadComplete.bind(obj),
+  })
+
+  evaporate.add(config)
+}
+
+// cloudfront, aws_url, awsRegion and s3Acceleration
+
+test.cb('should respect awsRegion and cloudfront when not defined', (t) => {
```

Comment:
Calling `test.cb` is not really needed for these cases. That's only when you want the test to be ended in a callback. Here you can just use `test(` and you don't need to pass no `t` or call `t.end()`.